syntax = "proto3";
package job_export;




import "google/protobuf/struct.proto";

/*
GetJobTableToolJson请求
*/
message GetJobTableToolJsonRequest {
    /*
    绑定 tool 作业的 Task id
    */
    string jobTaskIdTool = 1 ;
}

/*
GetJobTableToolJson返回
*/
message GetJobTableToolJsonResponse {
message TableDefs {
message Dimensions {
    /*
    列id
    */
    string id = 1 ;
    /*
    列标题
    */
    string name = 2 ;
}
        
message Columns {
    /*
    列id
    */
    string id = 1 ;
    /*
    列标题
    */
    string name = 2 ;
}
        
    
    /*
    输出表格id
    */
    string id = 1 ;
    /*
    输出表格标题
    */
    string name = 2 ;
    /*
    维度列
    */
    repeated Dimensions dimensions = 3 ;
    /*
    输出列
    */
    repeated Columns columns = 4 ;
}
        
    
    /*
    agents启动时间
    */
    google.protobuf.Struct startTime = 1 ;
    /*
    agents结束时间
    */
    google.protobuf.Struct endTime = 2 ;
    /*
    agents状态
    */
    google.protobuf.Struct status = 3 ;
    /*
    表格数据
    */
    google.protobuf.Struct tableData = 4 ;
    /*
    用户
    */
    string username = 5 ;
    /*
    工具的表格输出定义
    */
    repeated TableDefs tableDefs = 6 ;
    /*
    工具版本
    */
    string vId = 7 ;
    /*
    执行状态
    */
    string totalStatus = 8 ;
}

/*
GetJobTableToolJsonApi返回
*/
message GetJobTableToolJsonResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    job_export.GetJobTableToolJsonResponse data = 4 ;
}

